test/box-luatest/gh_10847_memtx_sort_data_test.lua

```diff
@@ -0,0 +1,766 @@
+        -- Create a snapshot and sort data file.
+        box.snapshot()
+
+        -- Remove all sort data files.
+        local fio = require('fio')
+        local glob = fio.pathjoin(box.cfg.memtx_dir, '*.sortdata')
+        local files = fio.glob(glob)
+        t.assert(#files > 0) -- a sort data file must have been created
+        for _, file in pairs(files) do
+            fio.unlink(file)
+        end
+    end)
+
+    -- Recover without a sort data file using the regular SK build.
+    cg.server:restart()
+    t.assert(cg.server:grep_log('memtx_sort_data_enabled = true but' ..
+                                ' no memtx sort data file found'))
+    t.assert(cg.server:grep_log('Space \'test\': done'))
+    t.assert(cg.server:grep_log('entering the event loop'))
+end
+
+-- Test that the sort data file is not recreated on box.snapshot after removal.
+g_generic.test_remove_sortdata_keep_snap = function(cg)
+    cg.server:exec(function()
+        -- Create a snapshot and sort data file.
+        box.snapshot()
+
+        -- Remove all sort data files.
+        local fio = require('fio')
+        local glob = fio.pathjoin(box.cfg.memtx_dir, '*.sortdata')
+        local files = fio.glob(glob)
+        t.assert(#files > 0) -- a sort data file must have been created
+        for _, file in pairs(files) do
+            fio.unlink(file)
+        end
+
+        -- Call box.snapshot() once again: sort data file not recreated.
+        box.snapshot()
+        t.assert_equals(#fio.glob(glob), 0)
+    end)
+end
+
+-- Test that checkpoint fails if a sort data file exists already.
+g_generic.test_remove_snap_keep_sortdata = function(cg)
+    cg.server:exec(function()
+        -- Remove all snapshots and sort data files.
+        local fio = require('fio')
+        local glob_snap = fio.pathjoin(box.cfg.memtx_dir, '*.snap')
+        local glob_sortdata = fio.pathjoin(box.cfg.memtx_dir, '*.sortdata')
+        local snapshots = fio.glob(glob_snap)
+        local sort_data_files = fio.glob(glob_sortdata)
+        t.assert(#snapshots > 0)
+        t.assert(#sort_data_files > 0)
+        for _, file in pairs(snapshots) do
+            fio.unlink(file)
+        end
+        for _, file in pairs(sort_data_files) do
+            fio.unlink(file)
+        end
+
+        -- Create a snapshot and sort data file.
+        box.snapshot()
+        t.assert(#fio.glob(glob_snap) == 1)
+        t.assert(#fio.glob(glob_sortdata) == 1)
+
+        -- Remove the .snap file.
+        fio.unlink(fio.glob(glob_snap)[1])
+        t.assert(#fio.glob(glob_snap) == 0)
+
+        -- Call to box.snapshot() fails: the sort data file exists already.
+        t.assert_error_msg_contains('.sortdata\': File exists', box.snapshot)
+        t.assert(#fio.glob(glob_snap) == 0)
+        t.assert(#fio.glob(glob_sortdata) == 1)
+    end)
+end
+
+-- Test that both snapshot and sort data file recreated if are absent.
+g_generic.test_remove_snap_and_sortdata = function(cg)
```

Comment:
AFAICS, there is no case checking that sortdata really gets garbage collected along with snapshot.